File: app/src/main/java/com/mss/imagesearcher/view/helpers/ImageLoader.java
package com.mss.imagesearcher.view.helpers;

import android.content.Context;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface ImageLoader {

    void load(@NonNull Context context,
              @NonNull String path,
              @NonNull ImageView imageView,
              @Nullable Runnable onSuccess,
              @Nullable Runnable onFailure);
}

File: app/src/main/java/com/mss/imagesearcher/view/helpers/ImageLoader.kt
package com.mss.imagesearcher.view.helpers

import android.content.Context
import android.widget.ImageView

interface ImageLoader {

    fun load(context: Context,
             path: String,
             imageView: ImageView,
             onSuccess: () -> Unit,
             onFailure: () -> Unit)
}
